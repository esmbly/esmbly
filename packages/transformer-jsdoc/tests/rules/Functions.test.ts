import { testRunner } from '../helpers/stringTestRunner';

describe('rule: Functions', () => {
  it('transforms an arrow function to TypeScript', async () => {
    const program = `
      /**
       * @param {number} n
       * @returns {number}
       */
      const square = (n) => n * n; 
    `;
    const [{ content }] = await testRunner(program);
    expect(content).toMatchSnapshot();
  });

  it('transforms a function declaration to TypeScript', async () => {
    const program = `
      /**
       * @param {number} n
       * @returns {number}
       */
      function square(n) {
        return n * n; 
      }
    `;
    const [{ content }] = await testRunner(program);
    expect(content).toMatchSnapshot();
  });

  it('transforms a function expression to TypeScript', async () => {
    const program = `
      /**
       * @param {number} n
       * @returns {number}
       */
      var square = function(n) {
        return n * n; 
      }
    `;
    const [{ content }] = await testRunner(program);
    expect(content).toMatchSnapshot();
  });

  it('transforms an exported function to TypeScript', async () => {
    const program = `
      /**
       * @param {number} n
       * @returns {number}
       */
      export function square(n) {
        return n * n; 
      }
    `;
    const [{ content }] = await testRunner(program);
    expect(content).toMatchSnapshot();
  });

  it('transforms a default exported function to TypeScript', async () => {
    const program = `
      /**
       * @param {number} n
       * @returns {number}
       */
      export default function square(n) {
        return n * n; 
      }
    `;
    const [{ content }] = await testRunner(program);
    expect(content).toMatchSnapshot();
  });

  it('adds the declare keyword and removes the implementation from functions tagged with @declare', async () => {
    const program = `
      /**
       * @param {number} n
       * @returns {number}
       * @declare
       */
      function square(n) {
        return n * n; 
      }
    `;
    const [{ content }] = await testRunner(program);
    expect(content).toMatchSnapshot();
  });
});
