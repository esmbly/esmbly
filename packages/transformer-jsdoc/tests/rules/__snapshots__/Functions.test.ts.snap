// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rule: Functions transforms a default exported function to TypeScript 1`] = `
"
      /**
       * @param {number} n
       * @returns {number}
       */
      export default function square(n: number): number {
        return n * n; 
      }
    "
`;

exports[`rule: Functions transforms a function declaration to TypeScript 1`] = `
"
      /**
       * @param {number} n
       * @returns {number}
       */
      function square(n: number): number {
        return n * n; 
      }
    "
`;

exports[`rule: Functions transforms a function expression to TypeScript 1`] = `
"
      /**
       * @param {number} n
       * @returns {number}
       */
      var square = function(n: number): number {
        return n * n; 
      }
    "
`;

exports[`rule: Functions transforms an arrow function to TypeScript 1`] = `
"
      /**
       * @param {number} n
       * @returns {number}
       */
      const square = (n: number): number => n * n; 
    "
`;

exports[`rule: Functions transforms an exported function to TypeScript 1`] = `
"
      /**
       * @param {number} n
       * @returns {number}
       */
      export function square(n: number): number {
        return n * n; 
      }
    "
`;
